counterfactuals/environment.py

```diff
@@ -0,0 +1,133 @@
+import babyai
+import pickle
+from gym_minigrid.minigrid import MiniGridEnv, OBJECT_TO_IDX, COLOR_TO_IDX
+import gym
+import numpy as np
+import random
+from dreamer.envs.wrapper import EnvWrapper
+from dreamer.envs.one_hot import OneHotAction
+from gym.spaces import Box
+
+class SlipperinessWrapper(EnvWrapper):
+    def __init__(self, env, slip_prob):
+        super().__init__(env)
+        self.slip_prob = slip_prob
+
+    def step(self, action):
+        if action == MiniGridEnv.Actions.forward: # Go forward action; no slippage for other actions
+            if np.random.uniform() < self.slip_prob:
+                action = random.choice(["left", "right"])
+                # By default, the agent can only move in a direction if it's facing that way.
+                # We can model slippage by turning a direction, moving that way, then turning back
+                # and only counting it as a single action.
+                self.step_count -= 2
+                if action == "left":
+                    super().step(MiniGridEnv.Actions.left)
+                    super().step(MiniGridEnv.Actions.forward)
+                    obs = super().step(MiniGridEnv.Actions.right)
+                else:
+                    super().step(MiniGridEnv.Actions.right)
+                    super().step(MiniGridEnv.Actions.forward)
+                    obs = super().step(MiniGridEnv.Actions.left)
+                return obs
+        return super().step(action)
+
+
+class OneHotObs(EnvWrapper):
+
+    def __init__(self, env):
+        super().__init__(env)
+        example_obs = self.reset()
+        obs_space = {}
+        for k, v in example_obs.items():
+            v_shape = v * 0
```

Comment:
Why do you multiply by 0 here?

Reply:
The observation space includes a .low element and a .high element.  Both of these are the same shape as the vector the observation will return, but they contain the lowest and highest values of the vector.  Since it's one-hot, we multiply by 0 to get the lowest value.  It's kinda confusing so I'll add a commend explaining.